### PR TITLE
feat(modelarts):add a new resource dataset version

### LINKED ISSUE #2118: [modelart] New resource support for dataset version
Hi team,

For the ModelArts service, I want a resource to manage the dataset version. Parameter support for resources is consistent with the console.

Best regards!

### Current Terraform Version

```
...
```

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add a new resource to manage dataset version

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #2118

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST='./huaweicloud/services/acceptance/modelarts' TESTARGS='-run=TestAccDatesetVersionResource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/modelarts -v -run=TestAccDatesetVersionResource_basic -timeout 360m -parallel 4
=== RUN   TestAccDatesetVersionResource_basic
=== PAUSE TestAccDatesetVersionResource_basic
=== CONT  TestAccDatesetVersionResource_basic
--- PASS: TestAccDatesetVersionResource_basic (100.61s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 100.754s
```

### CODE REVIEW

File: docs/resources/modelarts_dataset_version.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+# huaweicloud_modelarts_dataset_version
+
+Manages ModelArts dataset version resource within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+resource "huaweicloud_modelarts_dataset" "test" {
+  ...
```
Comment (niuzhenguo) [inline docs/resources/modelarts_dataset_version.md]: suggest to give a full dataset resource example here or just use a `dataset_id` variable.

File: docs/resources/modelarts_dataset_version.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
+subcategory: "AI Development Platform (ModelArts)"
+---
+
+# huaweicloud_modelarts_dataset_version
+
+Manages ModelArts dataset version resource within HuaweiCloud.
```
Comment (Jason-Zhang9309) [inline docs/resources/modelarts_dataset_version.md:7]: dataset is not a word

File: docs/resources/modelarts_dataset_version.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+* `description` - (Optional, String, ForceNew) Specifies the description of dataset version. It contains a maximum of
+ 256 characters and cannot contain special characters `!<>=&"'`. Changing this parameter will create a new resource.
+
+* `split_ratio` - (Optional, String, ForceNew) Specifies the ratio of splitting which randomly divides a labeled sample
+ into a training set and a validation set. Changing this parameter will create a new resource.
+
+-> Before you enable splitting, ensure each label has at least five labeled samples. Ensure there are at least two
+ multi-label samples, if any.
+
+* `hard_example` - (Optional, Bool, ForceNew) Specifies whether enable modelArts to write the hard example
```
Comment (Jason-Zhang9309) [inline docs/resources/modelarts_dataset_version.md]: s/modelArts/ModelArts

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_dataset_version_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	"github.com/chnsz/golangsdk/openstack/modelarts/v2/version"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func getDatesetVersionResourceFunc(config *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := config.ModelArtsV2Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmtp.Errorf("error creating ModelArts v1 client, err=%s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_dataset_version_test.go]: should we keep the style of error messge same with other resources?

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 32),
+					validation.StringMatch(regexp.MustCompile("^[\\-_A-Za-z0-9\u4e00-\u9fa5]+$"),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go]: this regular expression is not correct if the 'name' can only start with a letter as it's descibed in the doc

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+						"The name can only consist of letters, digits, Chinese characters, underscores (_)"+
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go]: keep the message same with the description in the doc

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+	client, err := config.ModelArtsV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ModelArts v2 client, err=%s", err)
+	}
+
+	datasetId := d.Get("dataset_id").(string)
+	versionId := d.Get("version_id").(string)
+	dErr := version.Delete(client, datasetId, versionId)
+	if dErr.Err != nil {
+		return diag.Errorf("error deleting ModelArts dataset version, ID= %s", d.Id())
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go:217]: return checkdeleted here could better

File: docs/resources/modelarts_dataset_version.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/modelarts_dataset_version.md]: done

File: docs/resources/modelarts_dataset_version.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/modelarts_dataset_version.md]: done

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go]: remove `start with` in doc.

File: huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/resource_huaweicloud_modelarts_dataset_version.go]: done

File: docs/resources/modelarts_dataset_version.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+* `description` - (Optional, String, ForceNew) Specifies the description of dataset version. It contains a maximum of
+ 256 characters and cannot contain special characters `!<>=&"'`. Changing this parameter will create a new resource.
+
+* `split_ratio` - (Optional, String, ForceNew) Specifies the ratio of splitting which randomly divides a labeled sample
+ into a training set and a validation set. Changing this parameter will create a new resource.
+
+-> Before you enable splitting, ensure each label has at least five labeled samples. Ensure there are at least two
+ multi-label samples, if any.
+
+* `hard_example` - (Optional, Bool, ForceNew) Specifies whether enable ModelArts to write the hard example
```
Comment (Jason-Zhang9309) [inline docs/resources/modelarts_dataset_version.md]: whether to

File: docs/resources/modelarts_dataset_version.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+  description = "Created by demo"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the resource. If omitted, the
+ provider-level region will be used. Changing this parameter will create a new resource.
```
Comment (ShiChangkuo) [inline docs/resources/modelarts_dataset_version.md]: please use two spaces for indentation, and check it in the whole file.

File: docs/resources/modelarts_dataset_version.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/modelarts_dataset_version.md:7]: AI domain term

File: docs/resources/modelarts_dataset_version.md (hunk 8 above)
Comment (hmbig2) [reply docs/resources/modelarts_dataset_version.md]: done

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_dataset_version_test.go [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+	"github.com/chnsz/golangsdk/openstack/modelarts/v2/version"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func getDatasetVersionResourceFunc(config *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := config.ModelArtsV2Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating ModelArts v1 client, err=%s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_dataset_version_test.go]: v1 or v2?

File: docs/resources/modelarts_dataset_version.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+* `files` - The total number of samples.
+
+* `storage_path` - The path to save the manifest file of the version.
+
+* `is_current` - Whether this version is current version.
+
+* `created_at` - The creation time, in UTC format.
+
+* `updated_at` - The last update time, in UTC format.
```
Comment (ShiChangkuo) [inline docs/resources/modelarts_dataset_version.md:80]: s/last/latest

File: docs/resources/modelarts_dataset_version.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 5 minute.
+
+## Import
+
+The dataset versions can be imported by `id`. It is composed of dataset ID and version ID, separated by a slash.
```
Comment (ShiChangkuo) [inline docs/resources/modelarts_dataset_version.md]: suggest to write like this:
The dataset versions can be imported by dataset ID and version ID, separated by a slash, e.g.

Comment (hmbig2) [reply docs/resources/modelarts_dataset_version.md]: ok

File: huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_dataset_version_test.go (hunk 10 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/modelarts/resource_huaweicloud_modelarts_dataset_version_test.go]: fix

File: docs/resources/modelarts_dataset_version.md (hunk 9 above)
Comment (hmbig2) [reply docs/resources/modelarts_dataset_version.md]: ok

File: docs/resources/modelarts_dataset_version.md (hunk 11 above)
Comment (hmbig2) [reply docs/resources/modelarts_dataset_version.md:80]: last update is correct

**Comment by hmbig2** [comment]
make testacc TEST='./huaweicloud/services/acceptance/modelarts' TESTARGS='-run=TestAccDatasetVersionResource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/modelarts -v -run=TestAccDatasetVersionResource_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasetVersionResource_basic
=== PAUSE TestAccDatasetVersionResource_basic
=== CONT  TestAccDatasetVersionResource_basic
--- PASS: TestAccDatasetVersionResource_basic (136.26s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts 136.354s
